### ISSUE TITLE
Establishing priorities for open-source contributors

### ISSUE THREAD

**Issue by tierninho** [desktop/desktop#7829]
Our open source contributors (OSC) have been invaluable thus far 💯 ✨ and they are a part of nearly every release -- which is fantastic! However, we are accustomed to the mentality of set-and-forget when it comes to labeling any issues with `help-wanted`. As this leaves an fairly ambiguous state of “what to work on next”, I am suggesting an improvement in the way we approach the labeling for these types of issues in order to reassure the OSC that they are working on the right things, to reinforce our current feature set with enhancements or bug fixes, and to ultimately ship a better product that maps back to our roadmap and our community.

#### The optimal flow for contributing would be as follows:
1. README link for those looking to contribute
2. The [Contribution](https://github.com/desktop/desktop/blob/development/.github/CONTRIBUTING.md) doc (with some edits)
3. A new project board with prioritized issues 
4. An actual issue

#### What needs working on?
Selecting what needs attention from the open-source community will be based on our roadmap and priorities, as well as the technical capabilities or inherent knowledge needed to complete the pull request with minimal guidance from core team.

#### The List
The list should be a living project board that is limited to 10 issues, prioritized from top down with a possible technical difficulty rating so first timers or those new to developing have a chance to contribute. The list should be reviewed weekly and hopefully we can assign one of our core developers as a sponsor to each issue. 

#### Next steps?
Just add your thoughts or give it a thumbs up/down and we will go from there. Open-source contributors feel free to add your thoughts too as this directly impacts you. Thank you!!

**Comment by shiftkey** [comment]
I'm still a fan of the label `help wanted` as it feels succinct and actionable, but 👍 to revisiting how we use it. I'd also love to ensure that issues we promote to this project board are detailed enough that they can be picked up by someone without needing to ask a lot of questions.

@tierninho have you given any thought to the size of task we'd use here? Smaller tasks are generally more approachable, and should result in a quicker review process, but larger ones are an opportunity to work more closely with the maintainers and deliver something more impactful, with the tradeoff being more time is required for reviews to ensure the work can be merged and released to users.

Basically I'd love to have a mix of smaller and larger tasks so that newer and experienced people can be more closely involved with the project.

**Comment by tierninho** [comment]
> I'm still a fan of the label help wanted as it feels succinct and actionable

No plans to remove the `help wanted` label as the issues will still be discoverable that way. Just the quantity of 'help-wanted` issues will be reduced to something manageable and directly tied to our roadmap.

> I'd also love to ensure that issues we promote to this project board are detailed enough that they can be picked up by someone without needing to ask a lot of questions.

💯 Yes, that is the idea as we to increase efficiency for all parties

>  have you given any thought to the size of task we'd use here?

I think the tasks will be evaluated as we move along, with most small and some a little more in-depth. It depends on what we aim to achieve at the time, but all tasks should have guidelines.

> Basically I'd love to have a mix of smaller and larger tasks so that newer and experienced people can be more closely involved with the project.

Agreed 👍

**Comment by tierninho** [comment]
>A new project board with prioritized issues

The current [project board]( https://github.com/desktop/desktop/projects/8) will be used for this.

### Next Steps

For the next iteration of this, I plan to solve two existing problems:

- Re-examine the existing `help-wanted` issues and close any stale or outlying issues. The definition of `help-wanted` may be eventually altered as it not confuse it with the new priority list.
- Create a new priority list for those issues that we specifically want the OSC to work on. This will include rolling up certain `help-wanted` issues into one, and aligning them with our roadmap and possibly our release cadence.
